### ISSUE TITLE
Integrate CUDA SP as a C++ wrapper 

### ISSUE THREAD

**Issue by breznak** [htm-community/nupic.cpp#50]
After https://github.com/htm-community/htm.cuda/issues/2

**Comment by dkeeney** [comment]
@breznak
I have no problem with cuda...or any other language, but if we are going to
move the interface for  Python (and eventually CSharp) to their own
repositories, shouldn't we also use a separate repository for cuda?

**Comment by breznak** [comment]
Good point, so either the code for languageX should stay `ifdef`'d, or in a separate repo.

**Comment by dkeeney** [comment]
I have a way to decouple the Python interface at the plugin interface such
that we should not need any ifdef's.

@chhenning wanted them to be in a separate repository.  I think perhaps a
sub cmake file would be ok but they need to be decoupled such that someone
that wants a CSharp or C++ only library would not also require Python to be
installed.
